### ISSUE TITLE
Incorrect error on type arguments

### ISSUE THREAD

**Issue by PizzaCrust** [intellij-rust/intellij-rust#6751]
## Environment

* **IntelliJ Rust plugin version:** 0.3.141.3634-203-nightly
* **Rust toolchain version:** 1.51.0-nightly (4d0dd02ee 2021-01-23) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2020.3.2 Community Edition (IC-203.7148.57)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
An error saying `wrong number of type arguments: expected 0, found 1` is shown on `A<B>` even though it compiles fine. 

## Steps to reproduce
```rust
struct A<const B: usize>;

impl<const B: usize> A<B> {

}
```
